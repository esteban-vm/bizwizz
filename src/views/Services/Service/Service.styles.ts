import clsx from 'clsx'

const styles = {
  wrapper: clsx(
    'flex w-80 flex-[1_1_100%] items-center justify-center',
    'overflow-hidden bg-lightgrey text-black',
    'hover:bg-primary hover:text-white',
    'pointer-coarse:bg-primary pointer-coarse:text-white',
    'lg:flex-[1_1_30%] [&>h4]:hover:translate-x-20',
    '[&>svg]:hover:scale-150 [&>svg]:hover:opacity-60',
    '[&>svg]:pointer-coarse:scale-100 [&>svg]:pointer-coarse:opacity-60'
  ),
  icon: clsx('mr-7 h-44 w-[4.8rem] scale-50 fill-primary-light opacity-0 transition-all duration-500'),
  title: clsx('select-none p-[3rem_6rem_3rem_0] text-[2.3rem] font-[500] transition-all duration-500'),
}

export default styles
