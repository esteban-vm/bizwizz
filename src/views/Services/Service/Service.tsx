import type { FC, ID, IconName } from '@/types'
import { Icon } from '@/components'
import styles from './Service.styles'

export type ServiceProps = {
  id: ID
  title: Capitalize<string>
  icon: IconName
}

const Service: FC<ServiceProps> = ({ id, title, icon }) => {
  return (
    <article className={styles.wrapper} id={id}>
      <Icon className={styles.icon} name={icon} />
      <h4 className={styles.heading}>{title}</h4>
    </article>
  )
}

export default Service
