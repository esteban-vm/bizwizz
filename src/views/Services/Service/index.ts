export { default, type ServiceProps } from './Service'
