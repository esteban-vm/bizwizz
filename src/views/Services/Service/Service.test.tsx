import { render, cleanup, within, screen } from '@/tests'
import Service from './Service'
import services from '../Services.data'

describe('Service:', () => {
  let asFragment: () => DocumentFragment
  const [testService] = services

  beforeEach(() => {
    void ({ asFragment } = render(<Service {...testService} />))
  })

  afterEach(cleanup)

  it('should render correctly', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should have correct structure', () => {
    const serviceElem = screen.getByRole('article')
    expect(serviceElem).toBeInTheDocument()
    expect(serviceElem).toHaveAttribute('id', testService.id)
    expect(within(serviceElem).getByRole('heading', { name: testService.title, level: 4 })).toBeInTheDocument()
    expect(within(serviceElem).getByRole('img', { hidden: true })).toBeInTheDocument()
  })
})
