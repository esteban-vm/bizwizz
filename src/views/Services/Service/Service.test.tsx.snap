// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Service: > should render correctly 1`] = `
<DocumentFragment>
  <article
    class="flex w-80 flex-[1_1_30%] items-center justify-center overflow-hidden bg-lightgrey text-black hover:bg-primary hover:text-white [&>h4]:hover:translate-x-20 [&>svg]:hover:scale-150 [&>svg]:hover:opacity-60"
    id="92cf33d6-5464-4bb5-8768-d62f81ab5036"
  >
    <svg
      aria-hidden="true"
      class="mr-7 h-44 w-[4.8rem] scale-50 fill-primary-light opacity-0 transition-all duration-500"
      role="img"
    >
      <use
        xlink:href="#icon-strategy"
      />
    </svg>
    <h4
      class="select-none p-[3rem_6rem_3rem_0] text-[2.3rem] font-[500] transition-all duration-500"
    >
      Digital Marketing
    </h4>
  </article>
</DocumentFragment>
`;
